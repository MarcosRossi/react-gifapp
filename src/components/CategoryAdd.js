import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CategoryAdd = function ({ setCategory }) {
  const [categoryName, setcategoryName] = useState('');
  const handleChange = (e) => {
    setcategoryName(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (categoryName.trim().length <= 2) return;

    setCategory((cats) => [categoryName, ...cats]);
    setcategoryName('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        value={categoryName}
        onChange={handleChange}
        type='text'
        placeholder='Insert a new  category name'
      />
    </form>
  );
};

CategoryAdd.propTypes = {
  setCategory: PropTypes.func.isRequired,
};
export default CategoryAdd;
