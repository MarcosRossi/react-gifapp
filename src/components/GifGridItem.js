import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, imgs: url }) => {
  return (
    <div className='card animate__animated animate__fadeInDown'>
      <p>{title}</p>
      <img src={url} alt={title} />
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  imgs: PropTypes.string.isRequired,
};
