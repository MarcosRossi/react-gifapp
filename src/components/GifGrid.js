import React from 'react';
import { GifGridItem } from '../components/GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category = 'goku' }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className='card-grid'>
        {images.map((img) => (
          <GifGridItem {...img} key={img.id} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
