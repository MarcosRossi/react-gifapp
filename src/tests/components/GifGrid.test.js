import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('tests in GifGrid.js', () => {
  const categoryName = 'Goku';

  test('should create the component ', () => {
    useFetchGifs.mockReturnValue({ data: [], loading: true });
    const wrapper = shallow(<GifGrid category={categoryName} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when images was loaded', () => {
    const data = [
      {
        id: 'abc',
        imgs: 'https://localhost/asd/asd.jpg',
        title: 'test',
      },
    ];
    useFetchGifs.mockReturnValue({ data, loading: false });
    const wrapper = shallow(<GifGrid category={categoryName} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBeFalsy();
    expect(wrapper.find('GifGridItem').length).toBe(data.length);
  });
});
