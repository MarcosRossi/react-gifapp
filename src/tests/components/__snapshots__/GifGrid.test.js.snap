// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in GifGrid.js should create the component  1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`tests in GifGrid.js should show items when images was loaded 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      imgs="https://localhost/asd/asd.jpg"
      key="abc"
      title="test"
    />
  </div>
</Fragment>
`;
